### DISCUSSION TITLE
Filtering Tag (JSON) via API

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#8331] [Archive]
Hi - 
potentially a very stupid question... but somehow I am not able to apply an API filter for Tags... before I had an M2O interface and things worked fine... then I figured out that there is a dedicated TAGS interface which is using a JSON field.

Displaying the infos in my web page via API works fine... BUT - when I want to filter based on Tags in my Tag Cloud I get error messages... neither "_contains" nor "_eq" work... error is that "THIS OPERATOR DOES NOT EXIST FOR JSON".

I am using PostgreSQL DB.

Any idea how I can fix this?

Thx a lot,
Roman

**Comment by rijkvanzanten** [comment]
> Any idea how I can fix this?

Postgres itself doesn't allow searching through JSON value using string like operators, like `=` or `LIKE %`, so I'm afraid there isn't much we can do about that at this moment

**Comment by domdus** [comment]
Had the same issue.
Change field type to CSV and use tags interface (is available there too).
Filtering works that way (Postgres here).

**Reply by rijkvanzanten** [reply]
Yup! A CSV saves as a CSV _string_, so all the regular strings operations are available 🙂

**Reply by ernst86913** [reply]
Thanks a lot to both of you… changed it to CSV for now and it works nicely… and hope that JSON support is improved over next months :)

**Reply by tutmoses** [reply]
Hello! Has this issue been solved? I'm struggling to filter Pages by parent_id because Collection Item dropdowns are forced to be JSON objects. I've tried loads of combinations, the latest being:

```
const subpagesFilter = JSON.stringify({
      "parent_id": { "key": { "_eq": page.id } },
      "status": { "_eq": "published" }
    });
```

But I keep getting this error:

```
{
      message: 'Invalid query. Invalid filter key "_eq" on "pages".',
      extensions: [Object]
    }
```

Is there a way around this?
